src/tests/mod.rs

```diff
@@ -0,0 +1,250 @@
+        path: Arc::new(path.clone()),
+    });
+
+    let avatar_settings = Data::new(AvatarSettings {
+        s3_bucket: String::from("test_avatar_bucket"),
+        retrieve_by_id_path: String::from("/avatar/get/id/"),
+        picture_api_url: String::from("http://localhost"),
+    });
+
+    let cis_client = Data::new(MockCisClient {});
+
+    let time_to_live = ::std::time::Duration::from_secs(60 * 60 * 24);
+    let cache = Data::new(RwLock::new(
+        LruCache::<String, String>::with_expiry_duration_and_capacity(time_to_live, 2000),
+    ));
+
+    // insert the cache entry for the current user
+    {
+        cache.write().unwrap().insert(
+            String::from("1"),
+            // random uuid4 without hyphens
+            String::from("78b814ab025e4da380836ff683be79e1"),
+        );
+    }
+
+    let app = App::new()
+        .wrap_fn(|req, srv| {
+            req.extensions_mut().insert(ScopeAndUser {
+                aa_level: AALevel::Medium,
+                groups_scope: GroupsTrust::Admin,
+                scope: if req.query_string().contains("@@testScope@@=authenticated") {
+                    Trust::Authenticated
+                } else {
+                    Trust::Public
+                },
+                user_id: String::from("1"),
+            });
+            srv.call(req)
+        })
+        .app_data(cis_client)
+        .app_data(loader)
+        .app_data(saver)
+        .app_data(avatar_settings)
+        .app_data(cache)
+        .service(
+            web::scope("/avatar")
+                .service(retrieve_app::<MockCisClient, FilesystemLoader>())
+                .service(send_app::<FilesystemSaver, FilesystemLoader>()),
+        )
+        .service(internal_send_app::<FilesystemSaver, FilesystemLoader>());
+    let mut app = test::init_service(app).await;
+
+    let sample_image_data = include_bytes!("data/sample_image.png");
+
+    let head = b"----abbc761f78ff4d7cb7573b5a23f96ef0\r\nContent-Disposition: form-data; name=\"file\"; filename=\"sample_image.png\"\r\nContent-Type: image/png\r\n\r\n";
+
+    let footer = b"\r\n----abbc761f78ff4d7cb7573b5a23f96ef0--\r\n\r\n";
+
+    let mut data = Vec::new();
+
+    data.extend(head.iter());
+    data.extend(sample_image_data.iter());
+    data.extend(footer.iter());
+
+    // construct payload
+    let payload = Bytes::from(data);
+
+    // dbg!(&payload);
+
+    let req = test::TestRequest::post()
+        .uri("/avatar/send/intermediate?@@testScope@@=authenticated")
+        .header(
+            "Content-Type",
+            "multipart/form-data; boundary=\"--abbc761f78ff4d7cb7573b5a23f96ef0\"",
+        )
+        .set_payload(payload)
+        .to_request();
+
+    #[derive(Deserialize, Debug)]
+    struct UuidResponse {
+        pub uuid: String,
+    }
+
+    let res_json: UuidResponse = test::read_response_json(&mut app, req).await;
+
+    // quick sanity check that the uuid is valid
+    assert_eq!(res_json.uuid.len(), 32);
```

Comment:
Just try to create a UUID from that https://docs.rs/uuid/0.8.1/uuid/struct.Uuid.html#method.parse_str